crates/context_aware_config/src/api/context/handlers.rs

```diff
@@ -248,10 +252,23 @@ fn update_override_of_existing_ctx(
         .filter(dsl::id.eq(&ctx.id))
         .select(dsl::override_)
         .first(conn)?;
-    cac_client::merge(&mut new_override, &ctx.override_);
+    cac_client::merge(
+        &mut new_override,
+        &Value::Object(ctx.override_.clone().into()),
+    );
     let new_override_id = hash(&new_override);
     let new_ctx = Context {
-        override_: new_override,
+        override_: Cac::<Overrides>::try_from_db(
+            new_override.as_object().cloned().unwrap_or(Map::new()),
+        )
+        .map_err(|err| {
+            log::error!(
+                "update_override_of_existing_ctx : failed to decode context from db {}",
+                err
+            );
+            unexpected_error!(err)
+        })?
+        .into_inner(),
```

Comment:
INFO: Validating the new map created after the merge operation perform [here](https://github.com/juspay/superposition/pull/264/files#diff-d134f3fb88f77342e5f42aaac3b57af7631de3f0926a1c0cd0e6b702d0e97c36R255)